router/dataplane_spec.gobra

```diff
@@ -67,10 +67,11 @@ pred (d *DataPlane) Mem() {
 	(d.internalNextHops != nil  ==> accAddr(d.internalNextHops))  &&
 	(d.svc != nil               ==> d.svc.Mem())                  &&
 	(d.macFactory != nil        ==> (
-		acc(d.key)                                        &&
-		acc(sl.Bytes(*d.key, 0, len(*d.key)), _) &&
-		scrypto.ValidKeyForHash(*d.key)                   &&
-		d.macFactory implements MacFactorySpec{d.key}))           &&
+		acc(d.key)                                      &&
+		acc(sl.Bytes(*d.key, 0, len(*d.key)), _)        &&
+		len(*d.key) > 0                                 &&
+		scrypto.ValidKeyForHash(*d.key)                 &&
+		d.macFactory implements MacFactorySpec{d.key})) &&
```

Comment:
the spacing for the last changed line should align with the outer `&&`s imo, as this is connecting to the conjuncts at that level

Reply:
Oh, I missed the parenthesis. It is now aligned.